app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,74 @@
+class TweetsController < ApplicationController
+
+	get '/tweets' do
+		if logged_in?
+			@tweets = Tweet.all
+			@user = User.find(session[:user_id])
+			erb :'/tweets/index'
+		else
+			redirect "/login"
+		end
+	end
+
+	post '/tweets' do
+    if params[:content] != ""
+      @tweet = Tweet.new(content: params[:content])
+      @user = User.find(session[:user_id])
+      @tweet.user_id = @user.id
+      @tweet.save
+      redirect to "/tweets/#{@tweet.id}"
+    else
+      redirect to '/tweets/new'
+    end
+  end
+
+	get '/tweets/new' do
+		if logged_in?
+      erb :'/tweets/new'
+    else
+      redirect to 'login'
+    end
+	end
+
+	# show page
+	get '/tweets/:id' do
+    if logged_in?
+      @tweet = Tweet.find(params[:id])
+      erb :'/tweets/show'
+    else
+      redirect to '/login'
+    end
+  end
+
+  # edit action
+  get '/tweets/:id/edit' do
+  	if logged_in?
+      @tweet = Tweet.find(params[:id])
+      erb :'/tweets/edit'
+    else
+      redirect to '/login'
+    end
+  end
+
+  patch '/tweets/:id' do
+    @tweet = Tweet.find(params[:id])
```

Comment:
you should do a second security check here to:

1: make sure a user is `logged_in?`
2. make sure that the `@tweet.user == current_user`